tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3792,7 +3795,8 @@ private[optimizer] abstract class OptimizerCore(
     val allocationSites = List.fill(allLocalDefs.size)(AllocationSite.Anonymous)
     val scope0 = optTarget.fold(Scope.Empty)(
         target => Scope.Empty.inlining((allocationSites, target)))
-    val scope = scope0.withEnv(OptEnv.Empty.withLocalDefs(allLocalDefs))
+    val scope1 = scope0.inlining(alreadyInlining)
+    val scope = scope1.withEnv(OptEnv.Empty.withLocalDefs(allLocalDefs))
```

Comment:
Create a set of things that are inlined first? Might be more readable:

```scala
val inlining = optTarget.fold(alreadyInlining)(target => alreadyInlining + (allocationSites, target))
val scope = Scope.Empty
  .inlining(inlining)
  .withEnv(...)
```

Reply:
Done.